data/transactions/verify/txn.go

```diff
@@ -566,3 +566,246 @@ func (w *worksetBuilder) next() (txnGroups [][]transactions.SignedTxn) {
 func (w *worksetBuilder) completed() bool {
 	return w.idx >= len(w.payset)
 }
+
+// UnverifiedTxnElement is the element passed to the Stream verifier
+// It represents an unverified txn whose signatures will be verified
+// BacklogMessage is a *txBacklogMsg from data/txHandler.go which needs to be
+// passed back to that context
+// Implements UnverifiedElement
+type UnverifiedTxnElement struct {
+	TxnGroup       []transactions.SignedTxn
+	BacklogMessage interface{}
+}
+
+// VerificationResult is the result of the txn group verification
+// BacklogMessage is the reference associated with the txn group which was
+// initially passed to the stream verifier
+type VerificationResult struct {
+	TxnGroup       []transactions.SignedTxn
+	BacklogMessage interface{}
+	Err            error
+}
+
+// NewBlockWatcher is a struct used to provide a new block header to the
+// stream verifier
+type NewBlockWatcher struct {
```

Comment:
Yes, I can move them to another file, like txnbatch.go. 

Moving them back to txn.go was not the intention, but moving them to streamverifier.go was essential to clearly show what was being abstracted in the current form.

Reply:
txnbatch looks good

Reply:
This was not possible, because these lines and the core functionality of preparing the batches were in the same file. 
I had to separate them, which meant either the core functionality lines were not to be moved, or these lines were not to be moved. These got moved :-(